src/braket/experimental/autoqasm/program/program.py

```diff
@@ -84,6 +89,22 @@ def __init__(self, oqpy_program: oqpy.Program, has_pulse_control: bool = False):
         self._oqpy_program = oqpy_program
         self._has_pulse_control = has_pulse_control
 
+    def bind_calibrations(self, gate_calibrations: Union[Callable, List[Callable]]) -> Program:
```

Comment:
A suggestion: this could be named `with_calibrations`, and it could return a new `Program` object (and a deep copy of the inner `oqpy.Program`) rather than modifying this instance of the program. I think this is possibly a more Python-friendly pattern, allowing you to do this:
```
program = my_prog()
program_cal1 = program.with_calibrations([cal1])
program_cal2 = program.with_calibrations([cal2])
```
so that `program_cal2` doesn't unexpectedly end up with `cal1` also attached.

Reply:
It actually does not modify the program object inplace. The name "bind" was confusing. Fixed.

Reply:
I lied, it was modifying the program inplace. I just didn't tested correctly. Fixed.